apps/dashboard/src/components/workflow-editor/steps/chat/chat-tabs.tsx

```diff
@@ -16,21 +17,36 @@ export const ChatTabs = (props: StepEditorProps) => {
   const isNovuCloud = !!(workflow.origin === WorkflowOriginEnum.NOVU_CLOUD && uiSchema);
   const isExternal = workflow.origin === WorkflowOriginEnum.EXTERNAL;
 
+  const { editorValue, setEditorValue, previewStep, previewData, isPreviewPending } = useEditorPreview({
+    workflowSlug: workflow.workflowId,
+    stepSlug: step.stepId,
+    controlValues: form.getValues(),
+  });
```

Comment:
Moved the state up the tree, to prevent re mounting when located inside of the child and persist state across tabs

Reply:
Ideally, this step should go even higher on the [ConfigureStep](https://github.com/novuhq/novu/blob/688fdd8058a79e383fefd0229074d56a28ac88c0/apps/dashboard/src/components/workflow-editor/steps/configure-step.tsx#L13) component so that we don't repeat it for every step component.

Reply:
I was thinking first of ConfigureStepTemplateForm, but the whole component structure was quite complex, and didn't want to get in to a full refactor there